translator/trace/zipkin/status_code.go

```diff
@@ -97,11 +98,25 @@ func (m *statusMapper) fromAttribute(key string, attrib *tracepb.AttributeValue)
 	return false
 }
 
-// attribToStatusCode maps an integer attribute value to a status code (int32).
-// The function return nil if the value is not an integer or an integer larger than what
-// can fit in an int32
-func attribToStatusCode(v *tracepb.AttributeValue) *int32 {
-	i := v.GetIntValue()
+// attribToStatusCode maps an integer or string attribute value to a status code.
+// The function return nil if the value is of another type or cannot be converted to an int32 value.
+func attribToStatusCode(attr *tracepb.AttributeValue) *int32 {
```

Comment:
What about returning `int32, error`? Do you think that is fine?

Reply:
Yes, that would be better. I just didn't want to spend more time on refactoring it assuming it needs to be converted to the new data model soon.

Reply:
What do you think?

Reply:
@bogdandrutu I converted the function as you suggested, there is bunch of logic downstream that uses pointers, I wouldn't touch that one in scope of this fix assuming we need to migrate to the new data model. Please take a look and let me know if this looks good now or better to switch the function back to pointers.